### PR TITLE
Mist

### PR DESCRIPTION
**Pull request by reedriley**
### Checklist
- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.

Additionally, when **adding a new cask**:
- [x] Checked there are no open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there are no closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

**Comment by reedriley** [comment]
While I think I've followed the token reference for naming this, part of me wonders whether it would be more clear/discoverable if it were named `ethereum-mist` instead.

File: Casks/mist.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask 'mist' do
+  version '0.8.4'
+  sha256 '50efcacfce0278269e8edd89aa91178c0ba00d9584fc019e25863939d0acf6ac'
+
+  url "https://github.com/ethereum/mist/releases/download/v#{version}/Mist-macosx-#{version.gsub '.', '-'}.dmg"
```
Comment (vitorgalvao) [inline Casks/mist.rb]: Instead of `#{version.gsub '.', '-'}` use `#{version.dots_to_hyphens}`

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
> part of me wonders whether it would be more clear/discoverable if it were named `ethereum-mist` instead

It would lead to duplicates (we know). The token reference exists exactly to prevent that.

**Comment by reedriley** [comment]
@vitorgalvao: So, your preference is `mist` for the package name?

**Comment by vitorgalvao** [comment]
Yes, as is.

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**